### ISSUE TITLE
Firestore 1.4.2 in Firebase 6.2.2: array-contains query returns empty array with offline persistence

### ISSUE THREAD

**Issue by henrylearn2rock** [firebase/firebase-js-sdk#1904]
### [REQUIRED] Describe your environment

  * Operating System version: Win10
  * Browser version: Chrome 75.0.3770.100
  * Firebase SDK version: 6.2.2
  * Firebase Product: firestore

#### Steps to reproduce:

Make a query with array-contains, no error from console but empty array returned.

#### Relevant Code:

```javascript

db
        .collection('Foo')
        .where('barArray', 'array-contains', 'something')
        


```

**Comment by attrib** [comment]
I can confirm this issue.

Downgrading to 6.2.1 (`npm install firebase@6.2.1 --save`) and the same code returns results.

**Comment by ahmedogela** [comment]
I have the same issue and I've tried v6.2.1 but it has the same issue too.
Note: When clearing the local storage it works for the first time then getting empty array when back to it again.

**Comment by jerryorta-dev** [comment]
I downgraded back to v6.2.0, the version I was using before this issue came up. It's still working in that version.

**Comment by schmidt-sebastian** [comment]
@henrylearn2rock Thank you for reporting this! We will take a look at this shortly. In the meantime, we recommend using an older version.

**Comment by var-const** [comment]
Unfortunately, we've had no luck trying to reproduce the issue so far, which seems to indicate there are some additional steps necessary for the bug to show up.

Any additional information would be most helpful. In particular:
* a minimal repro case will be super-useful if possible!
* can you please provide logs (please make sure they contain no confidential data) (here's how to enable Firestore logging)?
* does disabling persistence make any difference?
* is there anything interesting about your environment that could affect the results? In particular, are you using anything to package your JS code (say, Webpack)?

**Comment by henrylearn2rock** [comment]
a minimal repro case will be super-useful if possible!
> how, when it requires building up the collection with proper array index?

can you please provide logs (please make sure they contain no confidential data) (to enable Firestore logging, use firebase.firestore.setLogLevel('debug');, see docs)?

>  sure, but where can I send it to if we prefer not to post it here?

does disabling persistence make any difference?

> YES, disabling offline persistence works as expected.  Also, I can reproduce https://github.com/firebase/firebase-js-sdk/issues/1904#issuecomment-504743972.  I cleared the cache and the query returns the expected results for the first time, but it returns an empty array after a refresh.

is there anything interesting about your environment that could affect the results? In particular, are you using anything to package your JS code (say, Webpack)?

> YES, I use Webpack, but I also tried cdn at https://www.gstatic.com/firebasejs/6.2.2/firebase.js and encounter the same bug

**Comment by var-const** [comment]
Update: we have reproed the issue locally and are searching for the root cause.

**Comment by ahmedogela** [comment]
@henrylearn2rock

**Comment by hsubox76** [comment]
We just released v6.2.3 which should fix this problem.

**Comment by jerryorta-dev** [comment]
confirmed v6.2.3 is working in my environment. Thanks.

**Comment by henrylearn2rock** [comment]
v 6.2.3 still does **NOT** work for me.  It still returns empty array on refresh.

**Comment by henrylearn2rock** [comment]
Oh I just realized firestore is at 1.4.2

`yarn add v1.16.0
[1/4] Resolving packages...
Couldn't find any versions for "@firebase/firestore" that matches "1.4.3"`

Have you released "@firebase/firestore"?

**Comment by henrylearn2rock** [comment]
also confirmed v 6.2.3 is working now.  However @firebase/firestore is still at 1.4.2 on npm.

**Comment by hsubox76** [comment]
This was a rollback fix, so Firebase 6.2.3 should now have its `@firebase/firestore` dependency set to 1.4.1.  There is no Firestore 1.4.3 yet.

**Comment by var-const** [comment]
Release 6.2.3 has rolled back the regression. #1913 fixes the regression and will be a part of the next release.
